aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -353,14 +353,9 @@ inline Return Dispatcher::callWithDispatchKey(const TypedOperatorHandle<Return(A
       if (guard.needs_inputs) {
         std::vector<c10::IValue> stack;
         stack.reserve(sizeof...(Args));
-        auto boxed_all_args = impl::boxArgumentsOrCannotBoxIntoStack(stack, args...);
+        impl::boxArgumentsOrCannotBoxIntoStack(stack, args...);
 
         guard.before(op.schema().name(), stack, at::sequence_number::peek());
-
-        // if we could convert all the arguments, also pass the stack into the kernel call
-        if (boxed_all_args) {
-          return kernel.template callBoxedOrUnboxed<Return, Args...>(op, stack, std::forward<Args>(args)...);
```

Comment:
just curious, why did we decide to remove this branch?

Reply:
@ilia-cher sorry, meant to comment on this here myself, thanks for flagging! Eliminating `callBoxedOrUnboxed` saves a nontrivial amount of codesize - ~160k in libtorch.so release build. Separately, letting the control flow fall through to the unboxed `call` - rather than using `callBoxed` here and early returning - was more of a judgement call*, and one we can revisit after [complete boxing support](https://github.com/pytorch/pytorch/pull/42438/files#diff-355e2c36e6a2cc684d9341da5eed62f8R354) lands.

*about what percentage of the kernels we'll call from here will be unboxed